### ISSUE TITLE
Dependency to skyfield too heavy for just TEME-GCRS conversion

### ISSUE THREAD

**Issue by neothemachine** [skyfielders/python-skyfield#31]
I'm using sgp4 to get TEME coordinates. I could use skyfield to convert these to GCRS. My problem is that this adds a massive dependency to my own library, caused by the de421 dependency of skyfield  which is 26MB.

The main use case of skyfield probably requires de421 or similar to do its work, and the satellite ephemeris calculations are a side use case I guess. So removing de421 as dependency and letting the user install such package himself may not be what you want (loosing out-of-the box experience). On the other hand, letting the user have more freedom in choosing the package he wants might be good as well. After all, it's just `pip install skyfield de421` or `pip install skyfield de423` or similar.

What do you think?

**Comment by brandon-rhodes** [comment]
I am still thinking about the issue. The `de421` requirement was so that Skyfield could work and give answers out-of-the-box. Without an ephemeris, there is zero that Skyfield can do from its main API, because almost everything has to start with a planet. I am thinking about @tritium21’s idea of a `skyfield-core` that sophisticated users could select as their dependency if they did not care about Skyfield not working out of the box, and were happy to set up with another ephemeris instead.

**Comment by neothemachine** [comment]
You still didn't say whether users are allowed to use a different ephemeris than `de421`. Is that a use case of skyfield or not?

**Comment by barrycarter** [comment]
Thought: I believe the VSOP coefficients are shorter and wouldn't be as 
big a dependency as DE421 (or actually DE430 or DE431 these days). The 
older VSOP coefficients are fairly small, although VSOP2014 might be too 
large to make a difference. Additionally, the original VSOP coefficients 
are based on a different reference plane and center point, which may 
complicate things.

**Comment by brandon-rhodes** [comment]
@neothemachine — Sorry that I left that question hanging — here is the page in the docs about ephemerides:

http://rhodesmill.org/skyfield/planets.html

I will update the title of that section to say “Planets and Choosing an Ephemeris” so that it is more obvious from the Table of Contents that it talks about ephemerides — because right now it hides the subject by titling it just “The Planets.”

**Comment by neothemachine** [comment]
If I see it right then this all boils down to the `skyfield.api` module, which is a convenience module and the only piece of code having a dependency on `de421`.
If you split skyfield into skyfield-core and skyfield, then I think `skyfield.api` in its current state should not be part of skyfield-core because it would not be functional without `de421`. Of course you can say "so be it" and still include `skyfield.api` in skyfield-core and perhaps print an error when `de421` is not available. But somehow I don't like that because it could be cleaner. After all, `skyfield.api` is kind of an addon to skyfield-core.

So basically what I'm saying is: you could split skyfield into skyfield-core and skyfield, make `skyfield` a namespace package, and put `skyfield.api` into the new skyfield (meta)package. This is the cleanest way in my opinion.

**Comment by brandon-rhodes** [comment]
You will be happy to learn that, after some reflection on the future of Skyfield, I wound up agreeing with you! As you will see on the project home page, there is no longer any automatic dependency on any ephemeris — users have to be explicit (yay Zen of Python!) and ask for one to be downloaded:

```
from skyfield.api import load, now
planets = load('de421.bsp')
```

So you should now be able to use Skyfield on small machines without much memory as long as you just need the SGP4 routines and transforms. Enjoy, and feel free to open any further issues if you have a problem!

**Comment by neothemachine** [comment]
Looks good! I think that's the proper approach, definitely more maintainable, especially for you.
